File: Nodejs backend/services/roleMenuActionService.js
const RoleMenuAction = require('../models/RoleMenuAction');

const hasPermission = async (roleId, menuItemId, menuActionId) => {
  const permissions = await RoleMenuAction.find({
    role: roleId,
    menuItem: menuItemId,
    menuAction: menuActionId
  });

  if (!permissions.length) return false;
  return permissions.some(p => p.isAllowed);
};

module.exports = { hasPermission };

File: Backend/SanviiTechmet/src/main/java/com/nt/service/RoleMenuActionService.java
package com.nt.service;

import com.nt.entity.RoleMenuAction;
import com.nt.repository.RoleMenuActionRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class RoleMenuActionService {

    @Autowired
    private RoleMenuActionRepository roleMenuActionRepository;

    public boolean hasPermission(Integer roleId, Integer menuItemId, Integer menuActionId) {
        List<RoleMenuAction> roleMenuActions = roleMenuActionRepository
                .findByRoleIdAndMenuItemIdAndMenuActionId(roleId, menuItemId, menuActionId);

        return roleMenuActions.stream()
                .anyMatch(roleMenuAction -> roleMenuAction.isAllowed());
    }
}
